doc/releases/release-notes-4.1.rst

```diff
@@ -350,6 +350,31 @@ Libraries / Subsystems
 
 * Management
 
+  * hawkBit
+
+    * The hawkBit subsystem now uses the State Machine Framework internally.
+      The Kconfig symbols :kconfig:option:`CONFIG_SMF` and
+      :kconfig:option:`CONFIG_SMF_ANCESTOR_SUPPORT` are now required to be enabled to use the
+      hawkBit subsystem.
+
+    * The hawkBit subsystem now uses supports hawkBit instances, that run as a multi tenant system
+      and require the tenant to be configurable and don't use the default tenant ``default``. This
+      enables the use of some solutions, that require this. The Kconfig symbol
+      :kconfig:option:`CONFIG_HAWKBIT_TENANT` can be used to set the tenant name.
+
+    * The hawkBit subsystem now supports callbacks on specific events. The Kconfig symbol
+      :kconfig:option:`CONFIG_HAWKBIT_EVENT_CALLBACKS` can be used to enable this feature.
+      See .. doxygengroup:: hawkbit__event for more information or check out the source code of
+      :zephyr:code-sample:`hawkbit-api` for more information.
+
+    * The hawkBit subsystem now supports the save and restore of the download progress. When
+      enabled, the download progress is periodically saved to non-volatile storage. If a download
+      is interrupted, it can be resumed from the last saved point rather than starting over,
+      saving bandwidth and time. The Kconfig symbol :kconfig:option:`CONFIG_HAWKBIT_SAVE_PROGRESS`
+      can be used to enable this feature.
+      Use :kconfig:option:`CONFIG_HAWKBIT_SAVE_PROGRESS_INTERVAL` to set the interval in percent
+      when the progress is saved.
```

Comment:
I think this is too verbose for release notes -- why don't you make this part of the hawkbit docs instead, so that it "sticks" more. Those 4 pretty large paragraphs should be pretty much one sentence each instead, linking to relevant sections of the documentation

```
* Adopted the State Machine Framework internally to improve maintainability and extensibility (or just drop the "improviing..." part if it's too cheesy / unnecessary)
* Added support for multi-tenancy.
* Added support for registering callbacks on specific events
* Added support for storing partially downloaded update packages to non-volatile storage and resuming from the last save point.
```

Reply:
shortened it.